src/Hl7.Fhir.Base/Serialization/BaseFhirJsonDeserializer.cs

```
@@ -548,19 +636,24 @@ FhirJsonPocoDeserializerState state
                 // don't read any new data into the primitive here
                 reader.Read();
             }
+            else if(reader.TokenType == JsonTokenType.StartArray)
```

We should report an error here about nested lists not being allowed in FHIR.